api/restHandler/app/BuildPipelineRestHandler.go

```diff
@@ -304,6 +323,26 @@ func (handler PipelineConfigRestHandlerImpl) PatchCiMaterialSourceWithAppIdAndEn
 	common.WriteJsonResp(w, err, createResp, http.StatusOK)
 }
 
+func (handler PipelineConfigRestHandlerImpl) PatchCiMaterialSourceWithAppIdsAndEnvironmentId(w http.ResponseWriter, r *http.Request) {
+	bulkPatchRequest, userId, err := handler.parseBulkSourceChangeRequest(w, r)
+	if err != nil {
+		handler.Logger.Errorw("Parse error, PatchCiMaterialSource", "err", err, "PatchCiMaterialSource", bulkPatchRequest)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	token := r.Header.Get("token")
+	bulkPatchRequest.Token = token
+	// Here passing the checkAppSpecificAccess func to check RBAC
+	bulkPatchRequest.CheckAppSpecificAccess = handler.checkAppSpecificAccess
```

Comment:
any reason of passing this as part of request? This is not recommended as if this method is used again, this might get missed and technically also rbac is not part of request

Reply:
We have to check the RBAC based on ciPipelineMaterial Branch type that's why we have to send this inside the function.